esmvalcore/_data_finder.py

```diff
@@ -108,23 +188,87 @@ def get_start_end_year(filename):
     return int(start_year), int(end_year)
 
 
-def select_files(filenames, start_year, end_year):
-    """Select files containing data between start_year and end_year.
-
-    This works for filenames matching *_YYYY*-YYYY*.* or *_YYYY*.*
+def _parse_period(timerange):
+    start_date = None
```

Comment:
Could you add a docstring for this function and also for the other new functions that do not have one yet? A single line might be enough in many cases, but nothing at all seems a bit meagre. From our [contribution guidelines](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#what-should-be-documented):
> There is no need to write complete numpy style documentation for functions that are not visible in the [ESMValCore API Reference](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/api/esmvalcore.html#api) chapter on readthedocs. However, adding a docstring describing what a function does is always a good idea. For short functions, a one-line docstring is usually sufficient, but more complex functions might require slightly more extensive documentation.